server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -95,17 +97,47 @@ protected ClusterState execute(ClusterState currentState, AlterTableRequest requ
             .put(request.settings());
         settingsBuilder.remove(TableParameters.COLUMN_POLICY.getKey());
         Settings settings = settingsBuilder.build();
+
+        Metadata metadata = currentState.metadata();
+        RelationMetadata relation = metadata.getRelation(request.tableIdent());
+        if (request.partitionValues().isEmpty() && relation instanceof RelationMetadata.Table table) {
+            Metadata.Builder newMetadata = Metadata.builder(metadata);
+            Builder newSettings = Settings.builder()
+                .put(table.settings())
+                .put(settings);
+            for (String setting : settings.keySet()) {
+                if (!settings.hasValue(setting)) {
+                    newSettings.remove(setting);
+                }
+            }
+            newMetadata.addTable(
+                table.name(),
+                table.columns(),
+                newSettings.build(),
+                table.routingColumn(),
+                columnPolicy == null ? table.columnPolicy() : columnPolicy,
+                table.pkConstraintName(),
+                table.checkConstraints(),
+                table.primaryKeys(),
+                table.partitionedBy(),
+                table.state(),
+                table.indexUUIDs());
+            currentState = ClusterState.builder(currentState)
+                .metadata(newMetadata)
+                .build();
+        } else {
```

Comment:
Strictly speaking, this branch would be processed for non `RelationMetadata.Table` instances as well like `BlobTable`. Any reason to move this check?
```suggestion
        } else if(request.partitionValues().isEmpty() == false) {
```